### DISCUSSION TITLE
web版和纯前端版的区别是什么？

### DISCUSSION THREAD

**Discussion by helloyxw** [opensumi/core#1780] [Q&A] [answered]
web版和纯前端版的区别是什么？ 在功能上web版比纯前端版多了什么，纯前端版又少了什么功能？

**Comment by erha19** [comment, marked as answer]
@helloyxw  这里在功能的实现能力上面：

纯前端版本主要为一个在浏览器环境运行的静态资源站点，部署上只需要部署静态资源即可访问，仅需依赖一些开放或闭源实现的公共接口即可完成站点展示，如：[ext-fs-provider.ts#L34](https://github.com/opensumi/ide-startup-lite/blob/main/web-lite/file-provider/ext-fs-provider.ts#L34)，在功能上具备常规浏览器页面的基础能力，以及部分可在浏览器下运行的插件能力支持，参考 VS Code 中的 [Web Extensions](https://code.visualstudio.com/api/extension-guides/web-extensions)，在该类型的插件中能实现绝大多数语法高亮、命令执行等仅依赖浏览器环境即可实现的能力。

而 Web 版本则相较于纯前端版本，多处了一个需要部署 Node 进程的要求，支持全部类型的插件（包含 Web Extension，但没特殊要求情况下会有更优的替代插件），功能上也因此多出了许多浏览器中无法实现的能力，如本地文件读写、原生依赖调用等。
